libs/src/services/dataContracts/index.js

```diff
@@ -120,24 +120,30 @@ class AudiusContracts {
         UserReplicaSetManagerRegistryKey,
         this.getRegistryAddressForContract
       )
-      await this.UserReplicaSetManagerClient.init()
+      await this.UserReplicaSetManagerClient.init(selectNewEndpointOnRetry)
       if (this.UserReplicaSetManagerClient._contractAddress === '0x0000000000000000000000000000000000000000') {
         throw new Error(`Failed retrieve address for ${this.UserReplicaSetManagerClient.contractRegistryKey}`)
       }
     } catch (e) {
       // Nullify failed attempt to initialize
-      console.log(`Failed to initialize UserReplicaSetManagerClient`)
+      console.log(`Failed to initialize UserReplicaSetManagerClient with error ${e.message}`)
       this.UserReplicaSetManagerClient = null
     }
   }
 
   /* ------- CONTRACT META-FUNCTIONS ------- */
 
+  /**
+   * Retrieves contract address from Registry by key, caching previously retrieved data.
+   * Refreshes cache if cached value is empty or zero address.
```

Comment:
If you don't mind, pls add a comment on why a contract address would be zero ever <3

Reply:
yee added